main.tf

```diff
@@ -4,6 +4,8 @@ locals {
   enabled = module.this.enabled
 
   kms_key_id = local.enabled && var.encryption_enabled && var.kms_master_key_id != "" ? var.kms_master_key_id : ""
+
+  create_iam_policy = module.this.enabled && (length(var.allowed_aws_services_for_sns_published) > 0 || length(var.allowed_iam_arns_for_sns_publish) > 0)
```

Comment:
if any of these inputs are not present at plan time, this module will fail to provision